clients/cli/src/main/java/org/apache/gravitino/cli/commands/TagEntity.java

```diff
@@ -102,4 +110,33 @@ public void handle() {
 
     System.out.println(entity + " now tagged with " + all);
   }
+
+  private void checkTags(Table gTable) {
+    String[] associatedTags = gTable.supportsTags().listTags();
+    checkRedundantTags(associatedTags);
+  }
+
+  private void checkTags(Schema gSchema) {
+    String[] associatedTags = gSchema.supportsTags().listTags();
+    checkRedundantTags(associatedTags);
+  }
+
+  private void checkTags(Catalog gCatalog) {
+    String[] associatedTags = gCatalog.supportsTags().listTags();
+    checkRedundantTags(associatedTags);
+  }
+
+  private void checkRedundantTags(String[] associatedTags) {
+    List<String> redundantTags =
```

Comment:
There is no real need to do this extra work as you can just check for the exception. There is also the (rare and unlikely) possibility that a tag could be added after this check, as this is not in a transaction.